Project/webpack.config.js

```diff
@@ -1,26 +1,56 @@
 
 const CommunicationIdentityClient = require("@azure/communication-administration").CommunicationIdentityClient;
 const HtmlWebPackPlugin = require("html-webpack-plugin");
-const config = require("./config.json");
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const path = require('path');
 
-if(!config || !config.connectionString || config.connectionString.indexOf('endpoint=') === -1)
-{
-    throw new Error("Update `config.json` with connection string");
+let communicationIdentityClient = null;
+const getCommunicationIdentityClient = () => {
+    if (communicationIdentityClient != null) {
+        return communicationIdentityClient;
+    }
+    let config;
+    try {
+        config = require('./config.json');
+    } catch (error) {
+        return null;
+    }
+    const isValidConnectionString = config && config.connectionString && config.connectionString.indexOf('endpoint=') > -1; 
+    if (isValidConnectionString) {
+        communicationIdentityClient = new CommunicationIdentityClient(config.connectionString);
+        return communicationIdentityClient;
+    }
+    console.error("Cannot use custom Communication Service. Update `config.json` with connection string");
+    return null;
 }
 
-const communicationIdentityClient = new  CommunicationIdentityClient(config.connectionString);
+const env = process.env;
+env.development = false;
+
+const PORT = env.port || 9000;
 
-const PORT = process.env.port || 8080;
+const isProd = env.development == null || 
```

Comment:
You should not have to do so many checks to see if your in prod. One flag check should be enough to truly kno wether ur in prod or not.

Reply:
For example, a script can just passe flag "--env production", this tells your your going to build in prod and end up building in prod